automarker_2/automarker_app/lib/steps.py

```diff
@@ -626,9 +631,12 @@ def run(self, project_uri, clone_dir_path, self_test, config, fail_fast):
         if len(stderr):
             self.set_outcome(STEP_STATUS_ERROR, message=stderr)
         else:
-            command = (
-                f"{clone_dir_path/'automarker_venv'/'bin'/'pip'} install --upgrade pip"
-            )
+            if check_if_current_os_windows():
+                python_executable_path = 'Scripts/python'
+            else:
+                python_executable_path =  'bin/pip'
```

Comment:
Did you mean to put `pip` here? Also the naming can be better, or as a whole, why not have one function handle this, like:

```python
def get_python_executable_path():
    return 'Scripts/python' if is_current_os_windows() else 'bin/python'
```

then use that where we need to